File: LarkCardKit/Models/Card.cs
using System.Text.Json.Serialization;
using LarkCardKit.Config;
using LarkCardKit.Models.Elements;

namespace LarkCardKit.Models;

/// <summary>
/// 飞书卡片根模型
/// 表示一张完整的飞书卡片消息，符合飞书卡片 JSON 2.0 规范
/// </summary>
/// <example>
/// 以下示例演示如何创建卡片对象：
/// <code>
/// var card = new Card
/// {
///     Header = new CardHeader { Title = new PlainText { Content = "标题" } },
///     Body = new CardBody
///     {
///         Elements = { new PlainText { Content = "内容" } }
///     }
/// };
/// </code>
/// </example>
public class Card
{
    /// <summary>
    /// 卡片 JSON 结构版本
    /// 固定为 "2.0"，表示使用飞书卡片 2.0 规范
    /// </summary>
    [JsonPropertyName("schema")]
    public string Schema { get; set; } = "2.0";
    
    /// <summary>
    /// 卡片配置
    /// 包含卡片的更新策略、流式更新等配置项
    /// </summary>
    [JsonPropertyName("config")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public CardConfig? Config { get; set; }
    
    /// <summary>
    /// 卡片链接配置
    /// 设置点击卡片时的跳转链接
    /// </summary>
    [JsonPropertyName("card_link")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public object? CardLink { get; set; }
    
    /// <summary>
    /// 卡片头部
    /// 显示卡片的标题和图标，位于卡片顶部
    /// </summary>
    [JsonPropertyName("header")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public CardHeader? Header { get; set; }
    
    /// <summary>
    /// 卡片主体
    /// 包含卡片的所有内容元素（文本、按钮、表单等）
    /// </summary>
    [JsonPropertyName("body")]
    public CardBody Body { get; set; } = new();
    
    /// <summary>
    /// 降级配置
    /// 当客户端不支持卡片 2.0 时的降级显示内容
    /// </summary>
    [JsonPropertyName("fallback")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public object? Fallback { get; set; }
}

File: Python/lark_card_kit/models/card.py
"""Card model - the root model for Lark cards."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from .card_config import CardConfig
from .card_header import CardHeader
from .card_body import CardBody
from .card_link import CardLink
from .fallback import Fallback


@dataclass
class Card:
    """Root model for Lark card 2.0 messages.

    Represents a complete Lark card message conforming to the 2.0 specification.
    """

    schema: str = field(default="2.0", init=False)
    config: CardConfig | None = None
    card_link: CardLink | None = None
    header: CardHeader | None = None
    body: CardBody = field(default_factory=CardBody)
    fallback: Fallback | None = None

    def to_dict(self) -> dict[str, Any]:
        """Convert the card to a dictionary for JSON serialization.

        Returns:
            Dictionary representation of the card.
        """
        result: dict[str, Any] = {"schema": self.schema}

        if self.config:
            result["config"] = self.config.to_dict()
        if self.card_link:
            result["card_link"] = self.card_link.to_dict()
        if self.header:
            result["header"] = self.header.to_dict()
        if self.body:
            result["body"] = self.body.to_dict()
        if self.fallback:
            result["fallback"] = self.fallback.to_dict()

        return result

    def to_json(self, indent: int | None = None) -> str:
        """Convert the card to a JSON string.

        Args:
            indent: Number of spaces for indentation. None for compact output.

        Returns:
            JSON string representation of the card.
        """
        import json

        return json.dumps(self.to_dict(), ensure_ascii=False, indent=indent)